File: backend/src/main/java/com/scinar/giderlerim/service/KategoriService.java
package com.scinar.giderlerim.service;

import com.scinar.giderlerim.dto.request.KategoriGuncelleRequest;
import com.scinar.giderlerim.dto.request.KategoriOlusturRequest;
import com.scinar.giderlerim.dto.response.ApiResponse;
import com.scinar.giderlerim.dto.response.KategoriResponse;

import java.util.List;

public interface KategoriService {
    ApiResponse<List<KategoriResponse>> listele(Long kullaniciId);
    ApiResponse<KategoriResponse> olustur(Long kullaniciId, KategoriOlusturRequest request);
    ApiResponse<KategoriResponse> guncelle(Long kullaniciId, Long kategoriId, KategoriGuncelleRequest request);
    ApiResponse<Void> sil(Long kullaniciId, Long kategoriId);
}

File: packages/shared/services/kategoriService.ts
import { AxiosInstance } from 'axios';
import { ApiResponse } from '../types/api.types';
import { KategoriOlusturRequest, KategoriResponse } from '../types/kategori.types';

export function createKategoriService(client: AxiosInstance) {
  return {
    async listele(): Promise<ApiResponse<KategoriResponse[]>> {
      const response = await client.get<ApiResponse<KategoriResponse[]>>('/kategoriler');
      return response.data;
    },

    async olustur(data: KategoriOlusturRequest): Promise<ApiResponse<KategoriResponse>> {
      const response = await client.post<ApiResponse<KategoriResponse>>('/kategoriler', data);
      return response.data;
    },

    async guncelle(id: number, data: KategoriOlusturRequest): Promise<ApiResponse<KategoriResponse>> {
      const response = await client.put<ApiResponse<KategoriResponse>>(`/kategoriler/${id}`, data);
      return response.data;
    },

    async sil(id: number): Promise<void> {
      await client.delete(`/kategoriler/${id}`);
    },
  };
}

export type KategoriService = ReturnType<typeof createKategoriService>;
